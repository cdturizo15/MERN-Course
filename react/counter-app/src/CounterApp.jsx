import PropTypes from 'prop-types';
import { useState } from 'react';



const CounterApp = ({ value }) => {
    const handleSum = () => setCounter( counter + 1 )
    const handleReset = () => setCounter( value )
    const handleSubstract = () => setCounter ( counter - 1 )
    const [ counter, setCounter ] = useState( value );
    return (
    <div>
        <h1>Contador</h1>
        <h2 data-testid='value'>Valor: { counter }</h2>
        <button onClick={ handleSum }>+1</button>
        <button onClick={ handleSubstract }>-1</button>
        <button onClick={ handleReset }>Reset</button>
    </div>)
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;