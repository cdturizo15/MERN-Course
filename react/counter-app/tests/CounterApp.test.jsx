import CounterApp from '../src/CounterApp'
import { fireEvent, render, screen } from "@testing-library/react"


describe('Probando componente Counter ', () => {
    test('Probando match con snapshot', () => {
        const {container} = render( <CounterApp value={0}/> )
        expect(container).toMatchSnapshot();
    })
    test('Valor inicial 0', () => {
        render( <CounterApp value={0}/> )
        expect(screen.findByText('0')).toBeTruthy();
    })

    test('debe aumentar en uno la cuenta', () => {
        render( <CounterApp value={0}/> )
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByTestId('value').innerHTML ).toContain('1')
    })
    test('debe disminuir en uno la cuenta', () => {
        render( <CounterApp value={0}/> )
        fireEvent.click( screen.getByText('-1') )
        expect( screen.getByTestId('value').innerHTML ).toContain('1')
    })
    test('debe reiniciar la cuenta', () => {
        render( <CounterApp value={0}/> )
        fireEvent.click( screen.getByText('Reset') )
        expect( screen.getByTestId('value').innerHTML ).toContain('0')
    })
})