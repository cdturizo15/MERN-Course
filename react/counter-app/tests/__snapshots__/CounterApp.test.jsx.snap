// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente Counter  Probando match con snapshot 1`] = `
<div>
  <div>
    <h1>
      Contador
    </h1>
    <h2
      data-testid="value"
    >
      Valor: 
      0
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
